meson.build

```diff
@@ -0,0 +1,44 @@
+## Experimetal meson build system for dinit
+project(
+    'dinit', 
+    'cpp', 
+    version : '0.15.1',
+    license : 'Apache',
+    default_options : 'cpp_std = c++11')
+
+## Includes
+## We include `src/includes/` & `dasynq/include/` as defualt_incdir 
+default_incdir = include_directories('src/includes/', 'dasynq/include/')
+dinit_needed_files = ['src/dinit-main.cc',
+                      'src/dinit.cc',
+                      'src/load-service.cc',
+                      'src/service.cc',
+                      'src/proc-service.cc',
+                      'src/baseproc-service.cc',
+                      'src/control.cc',
+                      'src/dinit-log.cc',
+                      'src/run-child-proc.cc',
+                      'src/options-processing.cc',
+                      'src/dinit-env.cc']
+
+## Use -D_GLIBCXX_USE_CXX11_ABI?
+if get_option('use_new_abi').auto()
+# ToDo. We must verify libstdc++ version and set -D_GLIBCXX_USE_CXX11_ABI to 1 or 0
+elif get_option('use_new_abi').enabled()
+  add_global_arguments('-D_GLIBCXX_USE_CXX11_ABI=1', language : 'cpp')
+elif get_option('use_new_abi').disabled()
+  add_global_arguments('-D_GLIBCXX_USE_CXX11_ABI=0', language : 'cpp')
+endif
+
+## ToDo; its must handeled with `meson_options.txt` instead of force them
+## Use -Wall -flto -fno-rtti -fno-plt
+add_global_arguments('-Wall', language : 'cpp')
+add_global_arguments('-flto', language : 'cpp')
+add_global_arguments('-fno-rtti', language : 'cpp')
```

Comment:
RTTI is handled by the `cpp_rtti` builtin option, so you can just set that to `=false` in default_options.

Reply:
Yes; but i have a question: How can set `cpp_rtti` to auto? i mean its must be disabled in MacOS/FreeBSD & it can be enabled with Linux.
From dinit's BUILD:
https://github.com/davmac314/dinit/blob/b856c50c5a2dd1a8e34e9857603483485f779e65/BUILD#L44-L46

Reply:
Is it really mandatory?

Anyway you could do something like this:

```meson
default_options: [
    'cpp_rtti=@0@'.format(host_machine.system() == 'darwin' ? 'false' : 'true'),
]
```

Reply:
I think about that

Reply:
I writing `cpp_rtti` as follows:
`'cpp_rtti=@0@'.format(host_machine.system() == 'linux' ? 'true' : 'false'),`
but meson give error: 
`meson.build:2:0: ERROR: Unknown variable "host_machine".`